server/tserver/src/main/java/org/apache/accumulo/tserver/NativeMap.java

```diff
@@ -519,36 +520,25 @@ protected void finalize() throws Throwable {
     }
   }
 
-  private void _mutate(Mutation mutation, int mutationCount) {
+  private int _mutate(Mutation mutation, int mutationCount) {
 
     List<ColumnUpdate> updates = mutation.getUpdates();
     if (updates.size() == 1) {
       ColumnUpdate update = updates.get(0);
       singleUpdate(nmPointer, mutation.getRow(), update.getColumnFamily(), update.getColumnQualifier(), update.getColumnVisibility(), update.getTimestamp(),
-          update.isDeleted(), update.getValue(), mutationCount);
+          update.isDeleted(), update.getValue(), mutationCount++);
     } else if (updates.size() > 1) {
       long uid = startUpdate(nmPointer, mutation.getRow());
       for (ColumnUpdate update : updates) {
         update(nmPointer, uid, update.getColumnFamily(), update.getColumnQualifier(), update.getColumnVisibility(), update.getTimestamp(), update.isDeleted(),
-            update.getValue(), mutationCount);
+            update.getValue(), mutationCount++);
       }
-
     }
+    return mutationCount;
   }
 
   public void mutate(Mutation mutation, int mutationCount) {
-    wlock.lock();
```

Comment:
Is this write lock removal because it was unnecessary? (`mutate(List, int)` was re-grabbing the same lock)